### PR TITLE
Remove "Module" from daml-doc H1 headers

### PR DESCRIPTION
**Pull request by bame-da**
We are changing how we are managing the [ToC in the docs.daml.com repo](https://github.com/digital-asset/docs.daml.com/blob/main/docs/index/_toc.yml).
The `.. toctree` directives are replaced by a single yaml-based ToC file linked above. The way the pages are currently generated, [we need to override each page's title in the ToC to stop it showing "Module" over and over](https://github.com/digital-asset/docs.daml.com/blob/main/docs/index/_toc.yml#L928-L1007). This is annoying to keep in sync.

Instead, we'd like to just [use a glob](https://github.com/digital-asset/docs.daml.com/blob/globs/docs/index/_toc.yml#L640-L641).
That glob mechanism is even more valuable for Daml Finance docs where the set of pages is expected to change more frequently.

Since we can't override page titles in the ToC with globs, we need this change to not have "Module" repeated over and over in the ToC.

CHANGELOG_BEING
- Remove the word "Module" from the H1 page header of
- module pages and stop overriding the ToC headers to make
- the generated pages easier to use. 
CHANGELOG_END

### CODE REVIEW

**Review by brian-weir — APPROVED** [review]

**Review by akrmn — APPROVED** [review]
I think this makes sense, even outside of docs.daml.com the `Module ` prefix doesn't add much value. The reason CI failed is that we have a few golden tests on the output of `daml damlc docs`, but i can update those

File: compiler/damlc/daml-doc/src/DA/Daml/Doc/Render/Rst.hs [hunk 1]
```diff
@@ -40,10 +40,7 @@ renderRst env = \case
         ] ++
         [ T.concat
             [ "   "
-            , unModulename moduleName
-            , " <"
             , T.pack (moduleNameToFileName moduleName)
-            , ">"
```
Comment (akrmn) [inline compiler/damlc/daml-doc/src/DA/Daml/Doc/Render/Rst.hs:46]: @bame-da I don't understand the intention here, this removes the links from the `daml damlc docs`-generated TOC, which seems unrelated to the problem you're trying to solve.

Comment (akrmn) [reply compiler/damlc/daml-doc/src/DA/Daml/Doc/Render/Rst.hs]: I took the freedom to undo this change

Comment (bame-da) [reply compiler/damlc/daml-doc/src/DA/Daml/Doc/Render/Rst.hs]: No, please don't.

A toc entry of the format

`Custom title <doc path>`

is only needed if you want the text in the ToC to be different to the page header.

If you want the text in the ToC to just be the page header, you can just have `doc path` as the ToC entry. And that's my aim here.

Comment (akrmn) [reply compiler/damlc/daml-doc/src/DA/Daml/Doc/Render/Rst.hs]: I see now, thanks for explaining!

Comment (akrmn) [reply compiler/damlc/daml-doc/src/DA/Daml/Doc/Render/Rst.hs]: the part that worries me is that I'm not sure how this would interact with modules that have multiple components in their names e.g. `DA.Action`; I'll add some tests for that
